### DISCUSSION TITLE
addOpenChannel???  joinChannel???

### DISCUSSION THREAD

**Discussion by iwog3927** [storycraft/node-kakao#268] [Q&A] [answered]
안녕하세요.
좋은 자료 감사드립니다.
공유해주신 자료를 가지고 공부중에 있는데 문의 사항이 있어서 글을 남깁니다.

TalkOpenChannelList안에 ( TalkClient.channelList.open) joinChannel과 addOpenChannel이 있는데 혹시 차이점이 뭔지 알 수 있을까요?

js 코드를 보면 joinChannel 의 return 값으로 addOpenChannel을 호출하는데  addOpenChannel이 어떤 동작을 하는지 궁금합니다.

감사합니다.

**Comment by raravel** [comment]
joinChannel 은 해당 채널에 실제 입장. (~~님이 입장하셨습니다.) 이벤트를 전송하는 것이고, addOpenChannel은 클라이언트 (노드카카오)에 채널 리스트를 추가하는 걸로 보입니다.

**Comment by storycraft** [comment, marked as answer]
`addOpenChannel` 메서드는 인자로 주어진 오픈채널을 해당 `TalkOpenChannelList`에서 관리하는 채널 객체(`TalkOpenChannel`)로 추가 후 반환하는 메서드 입니다.

`joinChannel` 메서드는 인자로 주어진 오픈링크 id의 방에 참가하고 성공시 `addOpenChannel` 메서드를 호출해 관리되는 채널 객체를 결과로 반환합니다.
